### ISSUE TITLE
`do_predict` option of `TrainingArguments` - but no way to pass test set.

### ISSUE THREAD

**Issue by PhilipMay** [huggingface/transformers#8179]
The `TrainingArguments` class has the option to pass `do_predict=True`. The doc sais: "Whether to run predictions on the test set or not."

But there is no way to pass a test set to the trainer. At least I can not find it in the documentation...

Can you please clarify / fix this?

Many thanks
Philip

**Comment by sgugger** [comment]
The `do_predict` argument (like `do-train` and `do-eval` is not used by `Trainer`), just by the training scripts provided as examples.

Getting predictions on a test set is done with `trainer.predict(test_dataset)`.

**Comment by PhilipMay** [comment]
Should corresponding documentation be added?

**Comment by sgugger** [comment]
Sure, do you want to take a stab at it?

**Comment by PhilipMay** [comment]
@sgugger I can do a PR if you want. But...

... for me it smells like a design flaw when this is only for CLI usage and has no meaning for the "normal use".
Should we consider just removing it?

How would a documentation look like? _"This field is just a workaround for CLI value storage for the example code and has no meaning for normal usage."_?

**Comment by sgugger** [comment]
It's not `TrainerArguments` but `TrainingArguments`, so I don't see the problem with some of those arguments being only for CLI usage. Besides, removing them would break existing code so it would do more harm than good IMO.

For the documentation itself, it's not just a workaround. Something along the line of
```
This argument is not directly used by :class:`~transformers.Trainer`, it's intended to be used by your training/evaluation scripts instead. See the `example scripts <https://github.com/huggingface/transformers/tree/master/examples>`___ for more details.
```
would sound better.

**Comment by PhilipMay** [comment]
> ```
> This argument is not directly used by :class:`~transformers.Trainer`, it's intended to be used by your training/evaluation scripts instead. See the `example scripts <https://github.com/huggingface/transformers/tree/master/examples>`___ for more details.
> ```
> 
> would sound better.

PR has been created: #8270

**Comment by PhilipMay** [comment]
closing this since PR was merged
